packages/browser/src/provider.ts

```diff
@@ -0,0 +1,268 @@
+import { EventEmitter } from 'events';
```

Comment:
Suggest using SafeEventEmitter from Metamask to prevent issues with error handling of some Dapp contexts where it expects to operate like Metamask: https://github.com/MetaMask/safe-event-emitter

Reply:
I was basing this on minimizing changes to the existing implementation, which extends from `EventEmitter` directly.

Reply:
Ended up using it, might as well add some more safety here.